icepyx/core/read.py

```diff
@@ -387,14 +454,42 @@ def vars(self):
 
         if not hasattr(self, "_read_vars"):
             self._read_vars = Variables(
-                "file", path=self._filelist[0], product=self._prod
+                "file", path=self.filelist[0], product=self.product
             )
 
         return self._read_vars
+    
+    @property
+    def filelist(self):
+        """
+        Return the list of files represented by this Read object.
+        """
+        return self._filelist
+
+    @property
+    def product(self):
+        """
+        Return the product associated with the Read object.
+        """
+        return self._product
 
     # ----------------------------------------------------------------------
     # Methods
-
+    
+    @staticmethod
+    def _extract_product(filepath):
+        """
+        Read the product type from the metadata of the file. Return the product as a string.
+        """
+        with h5py.File(filepath, 'r') as f:
+            try: 
+                product = f.attrs['short_name'].decode()
+                product = is2ref._validate_product(product)
+            # TODO test that this is the proper error
```

Comment:
Highlighting this to-do item.

Reply:
Created a blank file to confirm this raises an error ✅